nixos/tests/mesos.nix

```diff
@@ -1,32 +1,91 @@
-import ./make-test.nix ({ pkgs, ...} : {
-  name = "simple";
+import ./make-test.nix ({ pkgs, ...} : rec {
+  name = "mesos";
   meta = with pkgs.stdenv.lib.maintainers; {
-    maintainers = [ offline ];
+    maintainers = [ offline kamilchm cstrahan ];
   };
 
-  machine = { config, pkgs, ... }: {
-    services.zookeeper.enable = true;
-    virtualisation.docker.enable = true;
-    services.mesos = {
-      slave = {
-        enable = true;
-        master = "zk://localhost:2181/mesos";
-        attributes = {
-          tag1 = "foo";
-          tag2 = "bar";
-        };
+  nodes = {
+    master = { config, pkgs, ... }: {
+      networking.firewall.enable = false;
+      services.zookeeper.enable = true;
+      services.mesos.master = {
+          enable = true;
+          zk = "zk://master:2181/mesos";
       };
-      master = {
-        enable = true;
-        zk = "zk://localhost:2181/mesos";
+    };
+
+    slave = { config, pkgs, ... }: {
+      networking.firewall.enable = false;
+      networking.nat.enable = true;
+      virtualisation.docker.enable = true;
+      services.mesos = {
+        slave = {
+          enable = true;
+          master = "master:5050";
+          dockerRegistry = registry;
+          executorEnvironmentVariables = {
+            PATH = "/run/current-system/sw/bin";
```

Comment:
@kamilchm In the test framework in `mesos_test.py` we set the command to `cat test.result`, but that'll only work if `cat` is accessible on `$PATH` (in the original source (which remains unchanged, btw), the default mesos agent path is the typical /bin:/usr/bin:...etc..., so on normal distros `cat` and such would be on `$PATH`). So, to keep our test simple (that is, to avoid patching in the full path to `cat` in `mesos_test.py`) we mimic how mesos would run on other distros by using `/run/current-system/sw/bin` in this test case.

Alternatively, we could set the default to `/run/current-system/sw/bin` (e.g. via the `executorEnvironmentVariables.PATH`) so stuff like `coreutils` are accessible by default, and then we could remove this line.

Thoughts?